components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/LegacyJobOrchestrator.cs

```diff
@@ -744,11 +762,13 @@ IJsonSerializer jsonSerializer
                         foreach (var assignedJob in _assignedJobs) {
                             if (entryJobId == assignedJob.Value.Job.Id) {
                                 found = _assignedJobs.Remove(assignedJob.Key, out _);
+                                _publisherDiagnosticInfo.Remove(assignedJob.Key, out _);
                                 break;
                             }
                         }
                         if (!found) {
                             found = _availableJobs.Remove(entryJobId, out _);
+                            _publisherDiagnosticInfo.Remove(entryJobId, out _);
```

Comment:
Since we should not have entries in `_publisherDiagnosticInfo` for unassigned jobs, this operation should not be necessary.

Reply:
removed